adaptivefiltering/execute.py

```diff
@@ -119,8 +137,7 @@ def apply_adaptive_pipeline(
         )
 
         # Write the filter into the output directory
-        # TODO: Change this filename from hash to the saved filename once it is implemented
-        save_filter(filter, os.path.join(output_dir, "{hash}.json"))
+        save_filter(filter, os.path.join(output_dir, f"{filter.title}.json"))
```

Comment:
I think people will often use spaces as part of the title, but I consider it unfortunate to have files with spaces. Maybe something like `.lower().replace(" ", "_")` or similar should be used.